bundles/config/org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs/src/main/java/org/eclipse/smarthome/config/discovery/usbserial/linux/sysfs/internal/PollingUsbSerialScanner.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs.internal;
+
+import static java.lang.Long.parseLong;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDeviceInformation;
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDiscovery;
+import org.eclipse.smarthome.config.discovery.usbserial.UsbSerialDiscoveryListener;
+import org.eclipse.smarthome.config.discovery.usbserial.linux.sysfs.internal.DeltaUsbSerialScanner.Delta;
+import org.eclipse.smarthome.core.common.ThreadPoolManager;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Modified;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link UsbSerialDiscovery} that implements background discovery by doing repetitive scans using a
+ * {@link UsbSerialScanner}, pausing a configurable amount of time between subsequent scans.
+ *
+ * @author Henning Sudbrock - initial contribution
+ */
+@Component(configurationPid = "discovery.usbserial.linux.sysfs.pollingscanner")
+public class PollingUsbSerialScanner implements UsbSerialDiscovery {
+
+    private final Logger logger = LoggerFactory.getLogger(PollingUsbSerialScanner.class);
+
+    private static final String THREAD_POOL_NAME = "usb-serial-discovery-linux-sysfs";
```

Comment:
Maybe I don't understand the issue here yet. Why should we give the thread pool here the same name as another thread pool? Wouldn't this be confusing, having two different thread pools with the same name?

During development, I had thought about using the thread pool from the `AbstractDiscoveryService` in this discovery as well; I did not do this as the `UsbSerialDiscoveryService` (which extends the `AbstractDiscoveryService`) does not use the thread pool. The thread pool is used in the `PollingUsbSerialScanner`, which is decoupled from the `UsbSerialDiscoveryService`.

Reply:
As far as I am aware, using the same name to get the thread pool means reusing and thus sharing the pool. So If you would also use `discovery` here your code would use the very same pool as the `AbstractDiscoveryService`.

You mention that your `PollingUsbSerialScanner` is decoupled from the `UsbSerialDiscoveryService`, but the `UsbSerialDiscoveryService` holds and instance of `UsbSerialDiscovery`, which at runtime might be a `PollingUsbSerialScanner`.

What I am trying to say is: This part of your code which wants to use a thread pool belongs to "discovery" tasks and thus I thought it would make sense to share the pool.

Maybe @sjka or @kaikreuzer can comment on this, whether it makes sense to share the "discovery" thread pool for all discovery services?